这会是渐进式应用程序 PWA 衰落的开始吗？ (#8284)
xitu/gold-miner: article/2021/is-it-the-beginning-of-the-end-for-pwas.md

> * 原文作者：[Mahdhi Rezvi](https://medium.com/@mahdhirezvi)
> * 译文出自：[掘金翻译计划](https://github.com/xitu/gold-miner)
> * 本文永久链接：[https://github.com/xitu/gold-miner/blob/master/article/2021/is-it-the-beginning-of-the-end-for-pwas.md](https://github.com/xitu/gold-miner/blob/master/article/2021/is-it-the-beginning-of-the-end-for-pwas.md)
> * 译者：[霜羽 Hoarfroster](https://github.com/PassionPenguin)

> > * 校对者：

Author: ```suggestion
> * 校对者：[Chorer](https://github.com/Chorer)、[Usualminds](https://github.com/Usualminds)
```

> # 这会是渐进式应用程序 PWA 衰落的开始吗？

Reviewer: ```suggestion
# 渐进式应用程序 PWA 开始衰落了吗？
```

> > Firefox 之后会有谁跟上来这么做？

Reviewer 2: ```suggestion
> Firefox 之后会有谁效仿？
```

![由 [Szabo Viktor](https://unsplash.com/@vmxhu?utm_source=medium&utm_medium=referral) 上传至 [Unsplash](https://unsplash.com?utm_source=medium&utm_medium=referral)](https://cdn-images-1.medium.com/max/11232/0*LmCaetpJHYbKh2bn)

> 渐进式 Web 应用程序正通过 Web 表现形式为我们提供类似于原生应用程序的体验，改变着我们对应用程序的理解。这是一种描述利用现代浏览器支持的新功能的应用程序的方式，包括 Service Worker 和 Web App Manifest，并且还允许用户无需考虑其原生的操作系统，将 Web 应用程序升级为渐进式 Web 应用程序。

Reviewer 2: ```suggestion
渐进式 Web 应用程序正通过 Web 形式为我们提供类似于原生应用程序的体验，改变了我们对应用程序的理解。这种方式让应用程序得以利用现代浏览器支持的新功能，包括 Service Worker 和 Web App Manifest，并且还允许用户无需考虑其原生的操作系统，将 Web 应用程序升级为渐进式 Web 应用程序。
```

Author: 话说这里的 web form 以及 Net 的 WinForm 的 Form 应该咋搞（

Reviewer 2: > 话说这里的 web form 以及 Net 的 WinForm 的 Form 应该咋搞（

这里的 web form 就是 web 形式，承载在 web 平台上的。winForm 我没了解过

> 但是，多年来，PWA 本身也带来了一些隐私问题，导致 Apple 停用了 Safari 上的一些 PWA 功能。你可以[阅读有关它们的更多信息](https://blog.bitsrc.io/the-darker-side-of-pwas-you-might-not-be-aware-of-ffa7b1d08888)。

Reviewer 2: ```suggestion
但是，PWA 在这些年也引发了一些隐私问题，导致 Apple 停用了 Safari 上的一些 PWA 功能。你可以在[另一篇文章中](https://blog.bitsrc.io/the-darker-side-of-pwas-you-might-not-be-aware-of-ffa7b1d08888)了解更多相关信息。
```

尽管如此，Chrome 以及 Firefox 等其他浏览器却仍然在大力支持 PWA 的开发。

> 不过最近，Mozilla 发布了旨在防止超级 Cookie 的 Firefox 85，似乎也已放弃了对桌面 PWA 的一项基本功能的支持，让我们看看它是什么：

Reviewer 2: ```suggestion
不过，Mozilla 最近发布了旨在防止超级 Cookie 的 Firefox 85，似乎也已放弃了对桌面 PWA 的一项基本功能的支持，让我们看看它是什么。
```

> ## 什么被舍弃了？

Reviewer 2: ```suggestion
## Firefox 放弃了什么？
```

> Firefox 放弃了一项用于支持在桌面上安装 PWA 的功能，**“特定站点的浏览器”（Site Specific Browser，SSB）**。

Reviewer 2: ```suggestion
Firefox 放弃了一项支持在桌面端安装 PWA 的功能，即**“特定站点的浏览器”（Site Specific Browser，SSB）**。
```

> 对于那些一直关注着这个问题的人，他们知道 Mozilla 曾经[提及过](https://bugzilla.mozilla.org/show_bug.cgi?id=1682593)他们可能会在将来的版本中放弃对 SSB 的支持。我们可以在 Issue Tracker 的注释中找到删除它的原因。

Reviewer 2: ```suggestion
一直关注这个问题的人可能会知道，Mozilla 曾经[提及过](https://bugzilla.mozilla.org/show_bug.cgi?id=1682593)未来的版本可能会放弃对 SSB 的支持。我们可以在 bug 跟踪板块的评论区中找到放弃这个功能的原因。
```
1. 两个“他们”可能引起歧义
2. comment 不是注释，是 bugzilla 这个网站的评论区（收集  bug 的，应该是 bug tracker 而不是 issue tracker）

Author: issue tracker 和 bug tracker 是同一个叭hhh

Reviewer 2: > issue tracker 和 bug tracker 是同一个叭hhh

我理解的是，issue 只是单纯表示应用存在问题，而问题的根源不一定是应用本身有 bug，可能是用户这边使用不当；而 bug 则表示是应用本身确实存在问题。所以 bug 应该算是 issue 的子集？另外这个 bugmozilla 的网站的板块也是写的 bug，所以这里使用 bug tracker 更合适（原文也是 bug tracker）。

> > SSB 功能一直都只能通过隐藏选项开启，并且有着多个已知的 bug。另外，从*用户调查*发现，几乎没有人意识到该功能的好处，因此目前我们没有继续开发该功能的想法。为了修复它的 bug，我们花费了过多的时间，持续维护的行为也让大家误以为我们还支持这个功能，因此我们决定将它从 Firefox 中移除。

Reviewer: ```suggestion
> SSB 功能一直都只能通过隐藏选项开启，并且有着多个已知的 bug。另外，从*用户调查*发现，几乎没有人意识到该功能的好处，因此目前我们没有继续开发该功能的想法。为了修复这个 bug，我们花费了过多的时间，持续维护的行为也让大家误以为我们还支持这个功能，因此我们决定将它从 Firefox 中移除。
```

> > SSB 功能曾经可通过隐藏的选项开启，并且有着多个已知的错误。另外，从*用户调查*发现，几乎没有人意识到该功能的好处，因此目前我们没有继续对其进行开发的想法。由于我们在这个功能上的错误的开发优先度，我们浪费了太多的时间，并且维护它的举动误导着大家，似乎告诉着别人我们还支持这个功能，因此我们决定将它从 Firefox 中删除。

Reviewer 2: ```suggestion
> SSB 功能一直都只能通过隐藏选项开启，并且有着多个已知的 bug。另外，从*用户调查*发现，几乎没有人意识到该功能的好处，因此目前我们没有继续开发该功能的想法。为了修复它的 bug，我们花费了过多的时间，持续维护的行为也让大家误以为我们还支持这个功能，因此我们决定将它从 Firefox 中移除。
```

### 什么是 SSB？

> SSB 是一项实验性功能，允许任何网站以其自己的窗口在桌面模式下运行。它在 Firefox 73 及更高版本中可用，让我们可以以最简的 UI 窗口中启动任何网站。

Reviewer 2: ```suggestion
SSB 是一项实验性功能，允许任何网站以其独有的窗口在桌面模式下运行。该功能在 Firefox 73 及更高版本中可用，让我们可以在 UI 精简的窗口中启动任何网站。
```
1. “其和“自己”重复了
2. 以最简的 UI 窗口中启动任何网站 => 在最简的 UI 窗口中启动任何网站

> ## 这是底线吗？

Reviewer 2: ```suggestion
## 这算是坏了一桩好事吗？
```

> 这完全取决于你的观点和你对 PWA 的预期用途。如果你认为 PWA 是获取独立于平台并像原生应用程序一样运行的应用程序的一种方式，那么你可能会对这一决定感到失望。但如果你认为 PWA 只是桌面上的快捷方式，那么你可能不会在意此更改。

Reviewer 2: ```suggestion
这完全取决于你的观点和你对 PWA 的预期用途。如果你认为 PWA 可以让应用独立于平台并像原生应用一样运行，那么你可能会对这一决定感到失望。但如果你认为 PWA 只是桌面上的快捷方式，那么你可能不会在意此更改。
```

> 有很多人对此决定不满意，但是与用户总数相比，这个数字还是很低的。这主要是因为这是实验性功能，并不为许多人所知。

Reviewer 2: ```suggestion
有很多人对此决定不满意，但是与用户总数相比，这部分人数还是很少的。主要原因在于这是实验性功能，并不为许多人所知。
```

正如[伊恩](https://www.i-programmer.info/news/87-web-development/14261-firefox-drops-support-for-pwa.html)所说：

> > 用户不会感到失望，因为他们不知道什么被拒绝了，以及这个举措原因是什么，并且会对你在创建原声应用程序上付出了额外的努力这个举动毫不在意，甚至感到疑惑。

Reviewer 2: ```suggestion
> 用户不会感到失望，因为他们不知道移除了什么功能。话说回来，他们又凭什么在乎你在创建原生应用程序上付出的额外努力呢？
```

> 由于技术发展日新月异，因此企业们在研发上花费了大量资源。实验功能至关重要，它们决定了应用程序的未来。当重要的东西被遗弃时，它会误导用户群体。

Reviewer 2: ```suggestion
由于技术发展日新月异，企业在研发上投入了大量资源。实验性功能至关重要，它们决定了应用程序的未来。当重要的功能被移除时，这会误导用户群体。
```

> 但是，当时机成熟且 PWA 成为突出的浏览器功能时，此举可能会让 Firefox 失去大量用户，让他们去使用替代的浏览器。

Reviewer 2: ```suggestion
但是，当时机成熟且 PWA 成为突出的浏览器功能时，此举可能会让 Firefox 失去大量用户，让他们转而使用其它的浏览器。
```

## 为什么要大惊小怪？

你可能会开始怀疑我是不是标题党，对吧！但不，我并不是。

> 科技界的共同趋势是：

Reviewer 2: ```suggestion
科技领域的共同趋势是：
```

> * 公司采取了决定性的举措

Reviewer 2: ```suggestion
* 公司采取了激进的举措
```

* 竞争对手嘲笑他们

> * 几年后，嘲笑此举的竞争对手也采取了同样的举动

Reviewer 2: ```suggestion
* 几年后，嘲笑此举的竞争对手也采取了同样的举措
```

> 你可以将其与一些值得注意的事件相关，例如取消耳机插孔，取消手机中的主页按钮以及最近的趋势 —— 取消掉充电器。

Reviewer 2: ```suggestion
你可以将这个趋势与一些值得注意的事件相联系，例如取消耳机插孔，取消手机中的主页按钮以及最近的趋势 —— 取消充电器。
```

> 提到的删除 SSB 的原因是，这个功能造成了麻烦，也浪费了宝贵的时间。在这种情况下，Chrome 和 Edge 等同类产品也可能会认为采取同样的行为对他们有利，并且由于过去发生过类似的事件，我们可能会在一段时间内丢失桌面端的 PWA。尽管这不太可能发生，但总是有机会发生这种情况，尤其是在科技领域。

Reviewer 2: ```suggestion
Mozilla 提到，移除 SSB 的原因是这个功能造成了麻烦，并且浪费了宝贵的时间。在这种情况下，Chrome 和 Edge 等同类产品也可能会认为采取同样的行为对他们有利，并且由于过去发生过类似的事件，我们可能会在一段时间内失去桌面端的 PWA。尽管这几乎不可能发生，但还是不排除有一定的几率，尤其是在科技领域。
```

highly unlikely：极不可能

> 你如何看待 Mozilla 的举动？在下面对其进行评论吧～

Reviewer 2: ```suggestion
你如何看待 Mozilla 的举措？在下面评论区分享你的看法吧。
```
